Sources/Diagnose/DiagnoseCommand.swift

```diff
@@ -101,26 +108,130 @@ public struct DiagnoseCommand: AsyncParsableCommand {
+      else {
+        continue
+      }
+      try? FileManager.default.copyItem(
+        at: fileUrl,
+        to: destinationDir.appendingPathComponent(fileUrl.lastPathComponent)
+      )
+    }
+    #endif
+  }
+
+  private func addSwiftVersion(toBundle bundlePath: URL) async throws {
+    print("-- Collecting installed Swift versions")
+
+    let outputFileUrl = bundlePath.appendingPathComponent("swift-versions.txt")
+    FileManager.default.createFile(atPath: outputFileUrl.path, contents: nil)
+    let fileHandle = try FileHandle(forWritingTo: outputFileUrl)
 
-    guard let reproducerBundle else {
-      print("No reducible crashes found")
-      throw ExitCode(1)
+    for toolchain in try await toolchainRegistry.toolchains {
+      guard let swiftUrl = toolchain.swift?.asURL else {
+        continue
+      }
+
+      try fileHandle.write(contentsOf: "\(swiftUrl.path) --version\n".data(using: .utf8)!)
+      let process = Process(
+        arguments: [swiftUrl.path, "--version"],
+        outputRedirection: .stream(
+          stdout: { try? fileHandle.write(contentsOf: $0) },
+          stderr: { _ in }
+        )
+      )
+      try process.launch()
+      try await process.waitUntilExit()
+      fileHandle.write("\n".data(using: .utf8)!)
     }
+  }
+
+  public func run() async throws {
+    print(
+      """
+      sourcekit-lsp diagnose collects information that helps developers of sourcekit-lsp diagnose and fix issues. 
+      This information contains:
+      - Crash logs from SourceKit
+      - Log messages emitted by SourceKit
+      - Versions of Swift installed on your system
+      - If possible, a minimized project that caused SourceKit to crash
+
+      All information will be collected into a folder on your computer.
+
+      The collection might take a few minutes.
+      ----------------------------------------
+      """
+    )
+
+    let date = ISO8601DateFormatter().string(from: Date()).replacingOccurrences(of: ":", with: "-")
+    let bundlePath = FileManager.default.temporaryDirectory
+      .appendingPathComponent("sourcekitd-reproducer-\(date)")
+    try FileManager.default.createDirectory(at: bundlePath, withIntermediateDirectories: true)
+
+    await orPrintError { try addCrashLogs(toBundle: bundlePath) }
+    await orPrintError { try await addOsLog(toBundle: bundlePath) }
+    await orPrintError { try await addSwiftVersion(toBundle: bundlePath) }
+    await orPrintError { try await addSourcekitdCrashReproducer(toBundle: bundlePath) }
+
     print(
       """
-        ----------------------------------------
-        Reduced SourceKit issue and created a bundle that contains a reduced sourcekitd request exhibiting the issue
-        and all the files referenced from the request.
-        The information in this bundle should be sufficient to reproduce the issue.
+      ----------------------------------------
+      Created a bundle that contains information about sourcekit-lsp issues encountered recently.
```

Comment:
`Bundle created. When filing an issue at https://github.com/apple/sourcekit-lsp/issues/new, please attach the bundle located at \(bundlePath.path).`